core/src/test/java/org/mskcc/cbio/portal/scripts/TestIntegrationTest.java

```diff
@@ -331,7 +351,37 @@ else if (gistic.getChromosome() == 10) {
         }
     }
 
-    
+    /**
+     * Test to confirm that 2 identical mutation events from 2 samples still point to the same
+     * mutation event despite unique constraint on `mutation_event`.
+     * @throws IOException
+     * @throws DaoException
+     */
+    @Test
+    public void testImportExtendedMutationDataExtendedWithSharedMutationEvent() throws IOException, DaoException {
+        // import maf
+        MySQLbulkLoader.bulkLoadOn();
+        File file = new File("src/test/resources/data_mutations_extended_duplicate_events.txt");
+        ImportExtendedMutationData parser = new ImportExtendedMutationData(file, testGeneticProfileId, null);
+        parser.importData();
+        MySQLbulkLoader.flushAll();
+        ConsoleUtil.showMessages();
+
+        // fetch mutations for test genetic profile
+        MutationMapperLegacy mutationMapperLegacy = applicationContext.getBean(MutationMapperLegacy.class);
+        List<String> geneticProfileStableIds = new ArrayList<String>();
+        geneticProfileStableIds.add("test_dup_mut_events");
+        List<Mutation> mutations = mutationMapperLegacy.getMutationsDetailed(geneticProfileStableIds,null,null,null);
+        // there are 2 identical mutation records from 2 different samples in test MAF
+        // verify that only one mutation event is associated with these samples
+        assertEquals(2, mutations.size());
+        Set<Integer> events = new HashSet<>();
+        for (Mutation mut : mutations) {
+            events.add(mut.getMutationEvent().getMutationEventId());
+        }
+        assertEquals(1, events.size());
```

Comment:
This test shows that mutation_event records *can* be reused. If we wanted to be more clear about when a mutation_event record should be reused and when a new distinct record should be created we would need to have several boundary cases showing when small variations allows a record to be reused or not. But at least this shows that when mutation profiles are imported, nearly identical records will share a mutation_event record (see notes below from data_mutations_extended_duplicate_events.txt for related comment)